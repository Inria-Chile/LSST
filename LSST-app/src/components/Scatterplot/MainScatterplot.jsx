import React, { Component } from 'react'
import Scatterplot from './Scatterplot';

class MainScatterplot extends Component {

    constructor(props){
        super(props);
    }

    render() {
        return (
            <div className="scatterplot">
                <Scatterplot displayedData={this.props.data} />
            </div>
            );
    }
}

export default MainScatterplot;
