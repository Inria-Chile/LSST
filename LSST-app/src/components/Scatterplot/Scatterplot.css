.whiteAxis text{
fill: #d4d7da;
}  

.whiteAxis line{
stroke: #d4d7da;
}

.whiteAxis path{
stroke: #d4d7da;
}

