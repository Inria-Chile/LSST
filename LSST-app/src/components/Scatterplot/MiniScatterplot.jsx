import React, { Component } from 'react'
import Scatterplot from './Scatterplot';


class MiniScatterplot extends Component {

    constructor(props){
        super(props);
        this.scatterplot=null;
    }

    setData(data){
        if(data && data.length > 0){
            this.scatterplot.setData(data);
        }
    }


  handleClick(){
    console.log("handling click")
    this.props.onScatterplotClick()
  }

    render() {
        return (
        <div className="scatterplot-container" onClick={ () => this.handleClick()}>
            <h3> Right Ascention / Declination </h3>
            <Scatterplot ref={instance => {this.scatterplot=instance;}} height={183}/>
        </div>
        );
    }
}


export default MiniScatterplot;
