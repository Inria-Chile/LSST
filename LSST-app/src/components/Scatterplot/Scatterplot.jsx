import React, { Component } from 'react'
import * as d3 from 'd3';
import ReactDOM from 'react-dom';
import './Scatterplot.css';


class Scatterplot extends Component {

    constructor(props){
        super(props);
        this.state={
            displayedData: null
        };
        this.data=null;
    }

    componentDidMount() {
        var dom = ReactDOM.findDOMNode(this);
        this.createScatterplot(dom, this.props);
    }

    componentDidUpdate() {
        var dom = ReactDOM.findDOMNode(this);
        this.removeScatterplot(dom);
        this.createScatterplot(dom, this.props);
    }

    createScatterplot(dom, props){
        let elem = ReactDOM.findDOMNode(this);
        let width = elem.offsetWidth;
        let height = this.props.height;

        let svg = d3.select(dom).append('svg').attr('class', 'd3').attr('width', width).attr('height', height);
        let margin = { top: 10, right: 15, bottom: 20, left: 25 };
        width = +svg.attr("width") - margin.left - margin.right;
        height = +svg.attr("height") - margin.top - margin.bottom;
        var g = svg.append("g").attr("transform", "translate(" + margin.left + "," + margin.top + ")");   

        var x = d3.scaleLinear().range([0, width]);
        var y = d3.scaleLinear().range([height,0]);
    
        if(this.state.displayedData==null){
            this.setData(this.props.data);
        }
        if(this.state.displayedData!=null){
            let data = this.state.displayedData;
            x.domain(d3.extent(data, function(d) { return d.fieldRA; }));
            let maxy = Math.abs(d3.max(data, function(d) { return d.fieldDec; }));
            let miny = Math.abs(d3.min(data, function(d) { return d.fieldDec; }));
            let ydom = maxy;
            if(miny > maxy) ydom = miny;
            y.domain([-ydom,ydom]);

            g.append("g")
            .attr("class", "axis axis--x")
            .attr("transform", "translate(0," + y.range()[0] / 2 + ")")
            .attr("class", "whiteAxis")
            .call(d3.axisBottom(x).ticks(10));

            g.append("g")
            .attr("class", "axis axis--y")
            .attr("transform", "translate(0,0)")
            .attr("class", "whiteAxis")
            .call(d3.axisLeft(y).ticks(10));
  
            g.selectAll("circle").data(data).enter().append("circle")
            .attr("class", "dot")
            .attr("r", 2)
            .attr("cx", function(d){
                return x(d.fieldRA);
            })
            .attr("cy", function(d){
                return y(d.fieldDec);
            })
            .style("fill", function(d) {        
            return "white"    
                });
        }   
    }

    removeScatterplot(dom){
        d3.select(dom).select('svg').remove();
    }

    setData(data){
        if(data && data.length > 0){
            this.data = data;
            this.setState({
                displayedData:data
            });
        }
    }
    
    setDisplayedDateLimits(start,end){
        let data = this.data;
        let dataToBeDisplayed = data.filter(function(d){
            
            return d.expDate >= start && d.expDate<= end;
        });
        this.setState({
            displayedData: dataToBeDisplayed
        });

    }

    render() {
        return(
        <div></div>);
    }
}

Scatterplot.defaultProps = {
    height: 300,
  };

export default Scatterplot;
