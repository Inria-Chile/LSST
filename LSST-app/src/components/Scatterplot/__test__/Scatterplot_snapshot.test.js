import React from 'react';
import Scatterplot from '../Scatterplot';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {mount} from "enzyme"; 
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });


describe('Scatterplot snapshot test',function(){
    let hola;

    it('Renders correctly',()=>{
        let displayedData = [
            {
                airmass: 1.3828,
                count: 1,
                expDate: 955592,
                expTime: 34,
                fieldDec: -40.227999999999994,
                fieldID: 166,
                fieldRA: 91.217,
                filterName: "z",
                lst: 1.0884,
                rotSkyPos: 0.375884,
                seeing: 1.00583,
                skybrightness_modified: 19.4018,
            }
        ]
        let tree = mount(<Scatterplot displayedData = {[]}/>);
        tree.setProps({displayedData : displayedData});
        expect(toJson(tree)).toMatchSnapshot();

    })

})