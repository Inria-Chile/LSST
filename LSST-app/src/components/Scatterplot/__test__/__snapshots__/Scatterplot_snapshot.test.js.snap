// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scatterplot snapshot test Renders correctly 1`] = `
<Scatterplot
  displayedData={
    Array [
      Object {
        "airmass": 1.3828,
        "count": 1,
        "expDate": 955592,
        "expTime": 34,
        "fieldDec": -40.227999999999994,
        "fieldID": 166,
        "fieldRA": 91.217,
        "filterName": "z",
        "lst": 1.0884,
        "rotSkyPos": 0.375884,
        "seeing": 1.00583,
        "skybrightness_modified": 19.4018,
      },
    ]
  }
  height={400}
>
  <div />
</Scatterplot>
`;
